pkg/frontend/readpath/queryfrontend/query_frontend.go

```diff
@@ -118,9 +119,14 @@ func (q *QueryFrontend) Query(
 
 		// If we need symbolization and this is a TREE query, convert it to PPROF
 		if shouldSymbolize && originalQuery.QueryType == queryv1.QueryType_QUERY_TREE {
+			maxNodes := int64(validation.SmallestPositiveNonZeroIntPerTenant(tenants, q.limits.MaxFlameGraphNodesMax))
+			if maxNodes == 0 {
+				maxNodes = int64(validation.SmallestPositiveNonZeroIntPerTenant(tenants, q.limits.MaxFlameGraphNodesDefault))
+			}
+
 			modifiedQueries[i].QueryType = queryv1.QueryType_QUERY_PPROF
 			modifiedQueries[i].Pprof = &queryv1.PprofQuery{
-				MaxNodes: 0,
+				MaxNodes: maxNodes,
```

Comment:
Shouldn't we take the `MaxNodes` from `queryv1.TreeQuery`?

Reply:
You're right. Also validation isn't needed here since it's already done in `ValidateMaxNodes`.